core/src/broadcast_stage/fail_entry_verification_broadcast_run.rs

```diff
@@ -31,44 +40,77 @@ impl BroadcastRun for FailEntryVerificationBroadcastRun {
         let bank = receive_results.bank.clone();
         let last_tick_height = receive_results.last_tick_height;
 
-        // 2) Convert entries to shreds + generate coding shreds. Set a garbage PoH on the last entry
-        // in the slot to make verification fail on validators
-        if last_tick_height == bank.max_tick_height() {
-            let mut last_entry = receive_results.entries.last_mut().unwrap();
-            last_entry.hash = Hash::default();
+        if bank.slot() != self.current_slot {
+            self.next_shred_index = 0;
+            self.current_slot = bank.slot();
         }
 
-        let next_shred_index = blockstore
-            .meta(bank.slot())
-            .expect("Database error")
-            .map(|meta| meta.consumed)
-            .unwrap_or(0) as u32;
+        // 2) If we're past SLOT_TO_RESOLVE, insert the correct shreds so validators can repair
+        // and make progress
+        if bank.slot() > SLOT_TO_RESOLVE && !self.good_shreds.is_empty() {
```

Comment:
qq: Is this right? It's comparing against the absolute `slot` with the constant `32`. So, the first condition  in this `if` clause is always `true` for almost all cases. I rather thought that some kind of _slot distance_ is intended.

Reply:
ah, **`Fail`**`EntryVerificationBroadcastRun`. Maybe, is this some test harness code?

Reply:
hehe yeah this is test harness code, used in local-cluster here: https://github.com/solana-labs/solana/blob/master/local-cluster/tests/local_cluster.rs#L1109